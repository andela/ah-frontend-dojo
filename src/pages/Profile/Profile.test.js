// react imports
import React from 'react';

// third party libraries
import { shallow } from 'enzyme';

// import profile
import {
  Profile,
  mapDispatchToProps,
} from 'pages/Profile';

const profile = {
  firstname: 'nadralia',
  lastname: 'lonerk',
  bio: 'this is my biodata',
  image: 'http://biodatainformation.jpg',
};

describe('Profile', () => {
  it('should render', () => {
    const component = shallow(<Profile profile={profile} />);
    expect(component).toMatchSnapshot();
  });

  it('should map dispatch to props', () => {
    const mockedDispatch = jest.fn();

    mapDispatchToProps(mockedDispatch).fetchProfile();
    expect(mockedDispatch).toHaveBeenCalled();
  });
});
