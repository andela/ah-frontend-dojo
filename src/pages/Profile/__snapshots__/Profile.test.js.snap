// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render 1`] = `
<div>
  <ToastContainer
    autoClose={4000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    role="alert"
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <Connect(Navbar) />
  <div
    className="profile-container"
  >
    <[object Object]
      fetchProfile={[Function]}
      profile={
        Object {
          "bio": "this is my biodata",
          "firstname": "nadralia",
          "image": "http://biodatainformation.jpg",
          "lastname": "lonerk",
        }
      }
    />
    <ProfileMain
      authenticatedUser="lou"
      fetchProfile={[Function]}
      match={[Function]}
      profile={
        Object {
          "bio": "this is my biodata",
          "firstname": "nadralia",
          "image": "http://biodatainformation.jpg",
          "lastname": "lonerk",
        }
      }
    />
  </div>
</div>
`;
